### PR TITLE
[RSDK-7133] Fix no msgs in queue bug after replug

### PR DESCRIPTION
**Pull request by hexbabe**
I would receive `"No message of type '{frame_type}' in frame queue.` error whenever I unplug and replug a camera in while having the stream open. This is the fix

### CODE REVIEW

File: src/worker.py [hunk 1]
```diff
@@ -110,21 +110,39 @@ def get_synced_msgs(self) -> Optional[Dict[str, BasePacket]]:
...
-        return self._get_most_recent_msg("depth")
+        for msg in q_snapshot:
+            self.add_msg(msg, frame_type, msg.get_sequence_num())
 
     def _get_most_recent_msg(
-        self, frame_type: Literal["color", "depth"]
+        self, q_handler: QueuePacketHandler, frame_type: Literal["color", "depth"]
     ) -> Optional[BasePacket]:
+        self._add_msgs_from_queue(frame_type, q_handler)
+        while len(self.msgs) < 1:
```
Comment (hexbabe) [inline src/worker.py:127]: This is the key line here. To wait until there is at least one message before trying to find a message on the queue

**Comment by hexbabe** [comment]
@seanavery bump on this review

File: src/worker.py [hunk 2]
```diff
@@ -233,16 +257,6 @@ def stop(self) -> None:
         self.manager.stop()
         self.oak.close()
 
-    def _add_msgs_from_queue(
```
Comment (seanavery) [inline src/worker.py:236]: Just changing location of this fn?

File: src/worker.py [hunk 3]
```diff
@@ -110,21 +110,39 @@ def get_synced_msgs(self) -> Optional[Dict[str, BasePacket]]:
...
         # Traverse in reverse to get the most recent
         for msg_dict in reversed(self.msgs.values()):
             if frame_type in msg_dict:
                 return msg_dict[frame_type]
         raise Exception(f"No message of type '{frame_type}' in frame queue.")
 
+    def get_most_recent_color_msg(
+        self, color_q_handler: QueuePacketHandler
+    ) -> Optional[BasePacket]:
+        return self._get_most_recent_msg(color_q_handler, "color")
```
Comment (seanavery) [inline src/worker.py:139]: Why are we proxying `get_most_recent_color_msg` to `_get_most_recent_msg`?

File: src/worker.py [hunk 4]
```diff
@@ -110,21 +110,39 @@ def get_synced_msgs(self) -> Optional[Dict[str, BasePacket]]:
...
+            self.add_msg(msg, frame_type, msg.get_sequence_num())
 
     def _get_most_recent_msg(
-        self, frame_type: Literal["color", "depth"]
+        self, q_handler: QueuePacketHandler, frame_type: Literal["color", "depth"]
     ) -> Optional[BasePacket]:
+        self._add_msgs_from_queue(frame_type, q_handler)
+        while len(self.msgs) < 1:
+            self._add_msgs_from_queue(frame_type, q_handler)
+            time.sleep(1)
```
Comment (seanavery) [inline src/worker.py:129]: Is it slow to sleep for a full second here?

**Review by seanavery — APPROVED** [review]

File: src/worker.py (hunk 3 above)
Comment (hexbabe) [reply src/worker.py:139]: v good point. I originally aliased it thinking there would be difference in logic but it didn't

File: src/worker.py (hunk 2 above)
Comment (hexbabe) [reply src/worker.py:236]: Yeah changing the call to be within `get_most_recent_msg`

**Pull request merged**